src/Numerics/Mesh/Metrics.jl

```diff
@@ -391,247 +574,151 @@ function computemetric!(
+            a22[i, j, k] =
+                ξ1x1[i, j, k, e] * ξ3x3[i, j, k, e] -
+                ξ1x3[i, j, k, e] * ξ3x1[i, j, k, e]
+            a23[i, j, k] =
+                ξ1x3[i, j, k, e] * ξ2x1[i, j, k, e] -
+                ξ1x1[i, j, k, e] * ξ2x3[i, j, k, e]
+            a31[i, j, k] =
+                ξ2x1[i, j, k, e] * ξ3x2[i, j, k, e] -
+                ξ2x2[i, j, k, e] * ξ3x1[i, j, k, e]
+            a32[i, j, k] =
+                ξ1x2[i, j, k, e] * ξ3x1[i, j, k, e] -
+                ξ1x1[i, j, k, e] * ξ3x2[i, j, k, e]
+            a33[i, j, k] =
+                ξ1x1[i, j, k, e] * ξ2x2[i, j, k, e] -
+                ξ1x2[i, j, k, e] * ξ2x1[i, j, k, e]
+
+            det =
+                ξ1x1[i, j, k, e] * a11[i, j, k] +
+                ξ2x1[i, j, k, e] * a12[i, j, k] +
+                ξ3x1[i, j, k, e] * a13[i, j, k]
+
+            x1ξ1[i, j, k, e] =
+                1.0 / det * (
+                    a11[i, j, k] * a11[i, j, k] +
+                    a12[i, j, k] * a12[i, j, k] +
+                    a13[i, j, k] * a13[i, j, k]
+                )
+            x1ξ2[i, j, k, e] =
+                1.0 / det * (
+                    a11[i, j, k] * a21[i, j, k] +
+                    a12[i, j, k] * a22[i, j, k] +
+                    a13[i, j, k] * a23[i, j, k]
+                )
+            x1ξ3[i, j, k, e] =
+                1.0 / det * (
+                    a11[i, j, k] * a31[i, j, k] +
+                    a12[i, j, k] * a32[i, j, k] +
+                    a13[i, j, k] * a33[i, j, k]
+                )
+            x2ξ1[i, j, k, e] =
+                1.0 / det * (
+                    a21[i, j, k] * a11[i, j, k] +
+                    a22[i, j, k] * a12[i, j, k] +
+                    a23[i, j, k] * a13[i, j, k]
+                )
+            x2ξ2[i, j, k, e] =
+                1.0 / det * (
+                    a21[i, j, k] * a21[i, j, k] +
+                    a22[i, j, k] * a22[i, j, k] +
+                    a23[i, j, k] * a23[i, j, k]
+                )
+            x2ξ3[i, j, k, e] =
+                1.0 / det * (
+                    a21[i, j, k] * a31[i, j, k] +
+                    a22[i, j, k] * a32[i, j, k] +
+                    a23[i, j, k] * a33[i, j, k]
+                )
+            x3ξ1[i, j, k, e] =
+                1.0 / det * (
+                    a31[i, j, k] * a11[i, j, k] +
+                    a32[i, j, k] * a12[i, j, k] +
+                    a33[i, j, k] * a13[i, j, k]
+                )
+            x3ξ2[i, j, k, e] =
+                1.0 / det * (
+                    a31[i, j, k] * a21[i, j, k] +
+                    a32[i, j, k] * a22[i, j, k] +
+                    a33[i, j, k] * a23[i, j, k]
+                )
+            x3ξ3[i, j, k, e] =
+                1.0 / det * (
+                    a31[i, j, k] * a31[i, j, k] +
+                    a32[i, j, k] * a32[i, j, k] +
+                    a33[i, j, k] * a33[i, j, k]
+                )
```

Comment:
@jkozdon @simonbyrne  I know that for a 3x3 we could have used a built-in method to invert the matrix, but I needed to do it for every quadrature point, in each element, so I went for the inversion of the matrix by hand. Let me know if you have any feedback, thanks!

Reply:
`inv` with StaticArrays should be equivalent.

Reply:
Thanks. Do you want me to change it?

Reply:
It's fine for now. We can revisit this in ClimateMachineCore.jl